### PR TITLE
enhancement(log): show stashes in log view, toggle visibility in view options popover

### PR DESCRIPTION
**Pull request by calahatras**

### CODE REVIEW

File: GitOut/Features/Git/LocalGitRepository.cs [hunk 1]
```diff
@@ -113,57 +113,25 @@ public async Task<IEnumerable<GitHistoryEvent>> LogAsync(LogOptions options)
...
-                            state = 1;
-                        }
-                        else
-                        {
-                            builder.BuildBody(line);
-                        }
-                        break;
+                    if (historyByCommitId.TryGetValue(item.ParentId!, out GitHistoryEvent? parent))
+                    {
+                        item.Branches.Add(GitBranchName.CreateLocal($"stash@{{{i}}}"));
```
Comment (mickedefault) [inline GitOut/Features/Git/LocalGitRepository.cs]: can we use the stash reflog selector as the branch name? [git pretty formats](https://git-scm.com/docs/pretty-formats)
>%gD
> reflog selector, e.g., refs/stash@{1} or refs/stash@{2 minutes ago}; [..]
> %gd
> shortened reflog selector; same as %gD, but the refname portion is shortened for human readability (so refs/heads/master becomes just master).

File: GitOut/Features/Git/LogOptions.cs [hunk 2]
```diff
@@ -2,11 +2,7 @@ namespace GitOut.Features.Git
 {
     public class LogOptions
```
Comment (mickedefault) [inline GitOut/Features/Git/LogOptions.cs]: can be converted to a record

**Review by mickedefault — CHANGES_REQUESTED** [review]
~unfortunately crashes at the moment when opening a project without stashes.~

File: GitOut/Features/Git/LocalGitRepository.cs (hunk 1 above)
Comment (calahatras) [reply GitOut/Features/Git/LocalGitRepository.cs]: You mean just prefix with `refs/`?

Comment (mickedefault) [reply GitOut/Features/Git/LocalGitRepository.cs]: hm. I can't remember why I figured that name would be better. 🤔 
what i was after was the descriptive name (for instance "WIP on <branch>: <description>")

Comment (calahatras) [reply GitOut/Features/Git/LocalGitRepository.cs]: The "WIP on" text is available with the latest change. I update GitStash to be a subclass of GitLogEntry instead, and read stashes using the same pretty format as logs, which means we get author, dates, and all that info as well.

File: GitOut/Features/Git/LogOptions.cs (hunk 2 above)
Comment (calahatras) [reply GitOut/Features/Git/LogOptions.cs]: It is now

**Review by mickedefault — APPROVED** [review]
:star:

**Pull request merged**